### ISSUE TITLE
A new physically based dust emission scheme with more aeolian physics

### ISSUE THREAD

**Issue by dmleung** [ESCOMP/CTSM#1604]
Hi this is Danny, a PhD student at UCLA working with Prof. Jasper Kok. We propose to add a new dust emission scheme improving the dust mobilization parameterizations in CESM2 (now based on 2.1 but I also tried 2.1.1, 2.1.3, and 2.2). We are collaborating with [Longlei Li](https://github.com/ESCOMP/CTSM/issues/1230) and Prof. Natalie Mahowald at Cornell University. Changes in files mainly include:

clm/src/biogeochem/DUSTMod.F90
clm/src/biogeophys/SoilStateType.F90
clm/src/biogeophys/SoilStateInitTimeConstMod.F90
clm/src/main/controlMod.F90
clm/src/main/clm_varctl.F90
cam/src/chemistry/modal_aero/dust_model.F90
clm/bld/namelist_files/namelist_definition_clm4_5.xml
user_nl_cam
user_nl_clm

Brief science information:
This is a scheme that builds upon issue [#1230](https://github.com/ESCOMP/CTSM/issues/1230) which switched CESM2's default dust emission scheme (Zender et al., 2003) to a more physical and less empirical one (Kok et al., 2014). Kok's scheme eliminates the need of an empirically tuned source function, and yields better agreement against observations over deserts both spatially and temporally. Based on #1230's edits, my modifications add new aeolian physics to the Kok's scheme, most notably, by adding the roughness effect (or called drag partition effect) which discounts surface soil erosion by winds due to the presence of local-scale land-surface roughness elements (mostly plants and rocks). We use a hybrid approach to account for both roughness from rocks (with a 2-D time-invariant dataset we provide) and roughness from plants (time-varying, as a function of CLM's LAI). We further include the dust emission intermittency effects due to boundary-layer turbulence. We note that these newly added physics can be applied to the default Zender's scheme too, and my code allows users to switch between the mobilization schemes they prefer.

--
Scientific description for code changes:

clm/src/biogeochem/DUSTMod.F90: 
1) Implement the new dust mobilization parameterization (Kok et al., 2014). 
2) Add the drag partition effect (or roughness effect) to reduce friction velocity for dust mobilization. 
3) Add the dust emission intermittency effect due to boundary-layer turbulence developed by Comola et al. (2019). 
Other changes include:
4) Add a new dust emission threshold scheme by Shao and Lu (2000) to replace the old Iversen and White (1982), but we allow users to choose which scheme they prefer.
5) Change the "optimal" soil particle diameter for dust mobilization to a realistic "median" soil particle diameter obtained from observed soil particle size distribution data. 
6) Eliminate the Owen effect which is less relevant to dust emission process.

cam/src/chemistry/modal_aero/dust_model.F90: Turn off the soil erodibility map (source function) used in the default Zender's scheme

clm/src/biogeophys/SoilStateType.F90
clm/src/biogeophys/SoilStateInitTimeConstMod.F90
clm/src/main/clm_varctl.F90 
clm/src/main/controlMod.F90
The above files are changed to read in the new time-invariant ncdf4 datasets needed to calculate the new physics. Three datasets included:
1) A static roughness dataset that contains rock roughness length for the roughness effect. 
2) A new clay fraction dataset compiled by SoilGrids for calculating soil moisture's effects to dust emission. CESM has its default dataset (from FAO) and users can choose which one they want. 
3) A land-cover dataset compiled from [Kobayashi et al. (2017)](https://www.researchgate.net/publication/317953457_Production_of_Global_Land_Cover_Data_-_GLCNMO2013) ([data link](https://globalmaps.github.io/glcnmo.html)). It quantifies the portion of rock cover over desert regions, which is useful for the roughness effect calculation.

clm/bld/namelist_files/namelist_definition_clm4_5.xml: Add flags to read the new namelist entries for reading the new ncdf4 files we provide.

user_nl_cam and user_nl_clm: Set the new namelist entries to read in the required datasets for simulation.

Long term goal:
We hope to add the suggested feature to the master branch of CESM2.

--
The code and case directories for all these changes are on Cheyenne.
I will also upload these files and datasets to my own Github repository later.

Thank you,
Danny

**Comment by wwieder** [comment]
Thanks for filing this issue @dmleung.  What's the timeline for opening a pull request (along with #1230) to bring these changes to the main development branch?  At one point we'd discussed doing these sequentially, with @L3atm's code coming in first to be followed by these changes?   Is this still the plan or are you going to bring this all in at one time?

**Comment by ekluzek** [comment]
This work is based off of the CESM release release-cesm2.1.0 and uses CLM version: release-clm5.0.14. The work is all done based on SourceMods and needs to be turned into a branch on github.

**Comment by ekluzek** [comment]
@dmleung the quick guide for working with git is here:

https://github.com/ESCOMP/CTSM/wiki/Quick-start-to-CTSM-development-with-git

**Comment by ekluzek** [comment]
@dmleung and I met and got him started at moving his work over into github. The clay soil texture part of this work is superseded by the work in #1303. The other two datasets are fundamental to this work. We will need to think a bit about how they will come into the model. In order to run at any resolution (the current sample datasets are only provided at f09 resolution).

**Comment by ekluzek** [comment]
We will also need to add more meta-data to the datasets, including some information about where the data came from, and what processing was done with it. There's a reference link for the LULC dataset, is there one for the roughness dataset? We also need to know if there is a tie in to PFT? So if PFT's change on the surface dataset should the new dust datasets be changed at the same time? Also ideally are these factors that change in time, or are they truly fixed in time? Or at least vary slow enough that time evolution doesn't matter?

**Comment by dmleung** [comment]
Hi Erik,
- We can provide the two datasets in higher resolutions (f05 or finer) if that helps.
- Sure, I can certainly re-create the ncdf4 datasets with more metadata and descriptions.
- The drag partition effect comes from the roughness dataset that Catherine Prigent provided upon our personal communication. I don't have any link to that dataset but the dataset comes from papers [here](https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2004JD005370) and [here](https://amt.copernicus.org/articles/5/2703/2012/amt-5-2703-2012.pdf). I think Catherine Prigent is helping the development of CLM roughness length parameterization as described [here](https://gmd.copernicus.org/preprints/gmd-2021-300/gmd-2021-300.pdf).
- To my knowledge, for now there is no science that explicitly and quantitatively links PFT changes to dust emission changes. My implementation does not relate PFT to my surface datasets or dust emission scheme, but I hope this piece of science will come in the future. To be honest the current science is still preliminary and don't know how to differentiate and parameterize between drag partition effects of different types of vegetation to dust emissions. 
- The time scale of impacts of PFT or LULC changes on dust emission changes will follow the time scales of PFT and LULC evolutions, i.e., 10+ years to decades. We think it is quite safe to provide a climatological representation of those surface datasets at least for present-day (2000s) simulations. But, yes, for historical (<1900) and future (>2050) simulations the climatological datasets will not be as accurate. We don't have these datasets in other time frames other than the present day.

**Comment by ekluzek** [comment]
Example user_nl_clm to exercise this:

```
 hist_fincl1='DSTFLXT'
 hist_fincl2='DSTFLXT','FIRE','C_d','LND_FRC_MBLE','GWC','WND_FRC_FT','WND_FRC_FT_DRY','LIQ_FRAC','WND_FRC_SOIL','ETA','USTAR','SSR','LAI','FRC_THR_RGHN_FCT','TLAI','TSAI','H2OSOI'
 hist_fincl3='DSTFLXT','FIRE','C_d','LND_FRC_MBLE','GWC','WND_FRC_FT','WND_FRC_FT_DRY','LIQ_FRAC','WND_FRC_SOIL','ETA','USTAR','SSR','LAI','FRC_THR_RGHN_FCT','TLAI','TSAI','H2OSOI'
 hist_fincl4='DSTFLXT','ETA'
 hist_fincl5='GWC','WND_FRC_FT_DRY','LIQ_FRAC','USTAR','FRC_THR_RGHN_FCT','LAI','SSR'
 hist_nhtfrq = 0, 0, -24, -1, -1
 hist_mfilt = 1, 1, 1, 24, 24
 clay_frc='/glade/u/home/dleung/CESM2/myinput/nas_cleared/f_clay_SoilGrids-09x125-12172021.nc'
 rough_fct='/glade/u/home/dleung/CESM2/myinput/nas_cleared/F_eff_r_static_mean_allLAI-09x125-12172021.nc'
 lulc_frc='/glade/u/home/dleung/CESM2/myinput/nas_cleared/GLCNMO_LULC_areafrc-09x125-12172021.nc'
```

**Comment by ekluzek** [comment]
Here's a comparison of the Prigent data as processed by Meier verses Leung. There are differences although Danny thinks the differences will be OK. An important difference though is that the Meier data is only over dry regions, which Danny did some processing (based on Catherine Prigents work) to make sure the data was provided everywhere.

[Comparing Ronny Meier and Danny Leung's roughness datasets.pdf](https://github.com/ESCOMP/CTSM/files/9294886/Comparing.Ronny.Meier.and.Danny.Leung.s.roughness.datasets.pdf)

**Comment by adamrher** [comment]
What's the status of this new dust scheme?

**Comment by dmleung** [comment]
Hi, thanks for asking! I think the issue #1897 has some updates from time to time. Erik Kluzek is coordinating with Francis Vitt to make sure the atm and lnd are communicating well, and our changes only apply to the upcoming CESM versions. The process is going on slowly, but @ekluzek should be able to give you more updates.

**Comment by adamrher** [comment]
Thanks. We are tuning up CESM3 and the dust emissions now. Should we expect this to come in before the code freeze next year @ekluzek? If you don't know, that's fine. I'm just trying to anticipate whether we should expect to have to retune the dust again.

**Comment by wwieder** [comment]
Erik can speak more about the details here, but I'd like to have Danny's PR in CESM3

**Comment by ekluzek** [comment]
This is dependent on #1836, which will have two PR's, one of which is #1967.

**Comment by ekluzek** [comment]
Our plan was just to move to the new scheme, and not to have a transition period away from the old. This does save some time in getting things setup to do both. This probably makes sense for the medium term, but it has issues with both the short and longer term. So here's why I think we should still invest in getting it setup so that we can use both. There will take some amount of time to allow both, but I don't think it's excessive.

- In the short term, while CESM3 is being tuned, it might be really useful to easily flip between Zender and Leung.
- While tuning for CESM3 we should expect that dust emissions will need to be tuned differently for Zender and Leung. It might be easier to tune Zender at the beginning while Leung is being figured out.
- Not allowing both means dust emissions will use the new scheme for clm4_5 and clm5_0
- By allowing a transition we can first bring in the new code as bit-for-bit and then turn it on in a separate step. This separates larger code changes, with a simple answer changing one. 
- The transition above also helps ensure that there aren't issues with code changes outside the main dust parameterization.
- Longer term, we have to realize that we want the ability to transition to newer parameters, so setting up the infrastructure to do this readily, is an investment in the future.
- By adding the infrastructure for it this way, we could if needed also add an option to run the Kok parameterization that this builds on. That would allow more flexibility in alternatives to the options for dust emissions.
- Longer term we expect that dust emissions will be updated and adding the infrastructure to easily update the parameterization makes it easier to do this.

I'll make this question in an issue and see if there are thoughts there on if we should do this or not.

**Comment by ekluzek** [comment]
@adamrher the rate we've been going at lately is something like 3-8 hours of my time every three weeks when I meet with @dmleung. @dmleung is done with his part of this. I've been working on doing some estimation for time left, and have been posting that to github issues. I've still got more work to do there. But, I can see that I need to run at a faster rate in order to get this done for CESM3. So @wwieder we should talk more about accommodating that.

**Comment by ekluzek** [comment]
@adamrher if you are starting to tune the dust with the Zender_2003 scheme at this point, is there any other key deadline that we should strive for to get Leung_2023 available for your use? I'm also wondering if you agree with me that for CESM3 tuning it would be best to enable being able to switch between Zender_2003 and Leung_2023 would be the best to accommodate doing tuning experiments? See above..

I've had the vague kind of desire to have this done by now, but I didn't do any planning to work out how to ensure to get it done. But, SWAG, wishes, and vague hopes don't make things happen. So I think it will help to do some planning on this.

**Comment by wwieder** [comment]
Thanks for all your thoughts on this @ekluzek   I thought when we met a while back with members of the CAM-Chem team they were not interested in maintaining the Zender scheme moving forward.  I appreciate this presents challenges for how to maintain older versions of the code base.  It's even more difficult for me to comment here, since the dust fluxes are a central topic for LMWG science.  I might suggest getting something that's useful for CAM developers to start evaluating that uses Leung et al. sooner rather than later.  Is this something that @slevis-lmwg could help with?

**Comment by adamrher** [comment]
> @adamrher if you are starting to tune the dust with the Zender_2003 scheme at this point, is there any other key deadline that we should strive for to get Leung_2023 available for your use? I'm also wondering if you agree with me that for CESM3 tuning it would be best to enable being able to switch between Zender_2003 and Leung_2023 would be the best to accommodate doing tuning experiments? See above..

I agree having the option to toggle between Zender and Leung is the ideal approach, especially if Zender is not available to us until late in the development cycle (worse case scenario of spring 2024, Id say). The sooner the better, but in this worse case scenario I think we could still tune up Zender in time if it comes in late -- my experience is that dust has a much more subtle effect on clouds/radiation than the big tuning parameters we currently use to tune the clouds/radiation, plus we have a new knob in microphysics that can dampen the impact to large dust changes on ice nucleation -- and so I'm fairly confident that we could make it work as the default dust scheme in CESM3 even if it comes in late. But it would be good to have the insurance, to toggle back to Leung just in case. 

@wwieder may have a different take on the latest acceptable date Zender can come in.

**Comment by wwieder** [comment]
If we aren't under a super tight deadline to get the Leung scheme in, but it can still be the default in CESM3 that sounds good to me.  Given @adamrher's comment it sounds like the capability to switch between the older Zender and newer Leung dust parameterizations is a good idea, @ekluzek.

**Comment by ekluzek** [comment]
Yes, @adamrher Zender_2003 is the old scheme and Leung_2023 is the new. @wwieder I'd love to have @slevis-lmwg help with this, and there are certainly parts that he could help with. We probably should discuss a plan for this soon to figure out how to make it happen for CESM3 and CAM tuning, and then we can update @adamrher and the CAM team about what our plan is. I think we should start talking about it Thursday, but figure on a longer planning session for just this work in the next few weeks.
